presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SimplifyExpressions.java

```diff
@@ -108,8 +112,9 @@ public Rewriter(Metadata metadata, SqlParser sqlParser, Session session, Map<Sym
         public PlanNode visitProject(ProjectNode node, RewriteContext<Void> context)
         {
             PlanNode source = context.rewrite(node.getSource());
-            Map<Symbol, Expression> assignments = ImmutableMap.copyOf(Maps.transformValues(node.getAssignments(), this::simplifyExpression));
-            return new ProjectNode(node.getId(), source, assignments);
+            expressionAssignments = node.getAssignments();
```

Comment:
You shouldn't be using class field to pass this information between `visit` methods. Use `context` instead.